core/domain/topic_jobs_one_off.py

```diff
@@ -165,3 +165,42 @@ def reduce(key, values):
                 sum(ast.literal_eval(v) for v in values))])
         else:
             yield (key, values)
+
+
+class RegenerateTopicSummaryOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to regenerate topic summaries."""
+
+    _DELETED_KEY = 'topic_deleted'
+    _PROCESSED_KEY = 'topic_processed'
+    _ERROR_KEY = 'topic_errored'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [topic_models.TopicModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            yield (RegenerateTopicSummaryOneOffJob._DELETED_KEY, 1)
+            return
+
+        try:
+            topic_services.generate_topic_summary(item.id)
+        except Exception as e:
+            yield (
+                RegenerateTopicSummaryOneOffJob._ERROR_KEY,
+                'Failed to create topic summary %s: %s' % (item.id, e))
```

Comment:
@aks681, actually, one comment. Does this hide the stacktrace of the actual error? (Can you check?)

If it does, please make sure to capture it, otherwise it'll be hard to debug the issue. Thanks!

Reply:
It does show the actual error since the exception message `e` is also present in the error message. The last test in the test file tests this: 

`for message in output:
   self.assertRegexpMatches(
                message,
                'object has no attribute \'canonical_story_references\'')`, 
where the expected message is the error message.

Reply:
Wait, that's not what I meant. I know that this shows the error message but my question is about showing the *stack trace* -- i.e. the line numbers where the exception happened. Could you PTAL again?

Reply:
Hm I also wonder whether we can do this more generally so that we don't need to do something custom each time. Maybe worth investigating approaches like this: https://www.datadoghq.com/blog/python-logging-best-practices/#python-exception-handling-and-tracebacks ?

Reply:
Done, used 'logging.exception' as that prints the stacktrace as well according to the link above.

Reply:
Hm I don't see logging.exception in that link (using Ctrl+F). @aks681, could you please point me to where it is?

Reply:
Actually, scratch that. Other sources suggest it works, so LGTM!